dashboard/scripts/archive/block_docs_to_i18n

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env ruby
+
+# Script to grab parameter information from block documentation .md files
+# and write it out to a .json localization file, following a key format
+
+require 'json'
+
+CDO_ROOT = '../../../'
+
+FunctionInfo = Struct.new(:name, :params)
+ParameterInfo = Struct.new(:name, :description)
+
+# @param [Array.<String>] doc_files
+# @return [Array.<FunctionInfo>] function infos
+def parse_function_docs(doc_files)
+  functions = []
+
+  doc_files.each do |doc_filename|
+
+    collecting_params = false
+
+    block_name = doc_filename.split('/').last.split('.').first
+    function = FunctionInfo.new(block_name, [])
+    functions.push(function)
+
+    File.readlines(doc_filename).each do |line|
+      if line.include? '[parameters]'
+        collecting_params = true
+        next
+      elsif line.include? '[/parameters]'
+        collecting_params = false
+        next
+      end
+
+      if collecting_params && line.match(/^\|[^-]/) && !line.include?('| Name')
+        param_name = line.split('|')[1].strip
+        param_name_no_brackets = param_name.tr('[]', '')
+        param_description = line.split('|')[4].strip
+        param_info = ParameterInfo.new(param_name_no_brackets, param_description)
+        function.params.push(param_info)
+      end
+    end
+  end
+  functions
+end
+
+def update_docs_i18n_file(functions, i18n_filename)
+  new_strings = {}
+
+  functions.each do |f|
+    f.params.each_with_index do |param, i|
+      new_strings["dropletBlock_#{f.name}_param#{i}"] = param.name
+      new_strings["dropletBlock_#{f.name}_param#{i}_description"] = param.description
+    end
+  end
+
+  json_file = i18n_filename
+  data_hash = File.exist?(json_file) ? JSON.parse(File.read(json_file)) : {}
+
+  modified_data_hash = data_hash.merge(new_strings)
+  sorted_data_hash = Hash[modified_data_hash.sort]
+
+  File.write(json_file, JSON.pretty_generate(sorted_data_hash) + "\n")
+end
+
+doc_files = Dir.glob("#{CDO_ROOT}pegasus/**/applab/docs/*.md").select { |e| File.file? e }
+puts "Processing #{doc_files.count} doc files..."
+function_infos = parse_function_docs(doc_files)
+puts "Updating i18n lines..."
+update_docs_i18n_file(function_infos, "#{CDO_ROOT}apps/i18n/common/en_us.json")
```

Comment:
maybe consider catching exceptions thrown by parse_function_docs and printing an error message with the filename?